src/interdiff.c

```diff
@@ -58,6 +59,27 @@
 #define PATCH "patch"
 #endif
 
+#define COLOR_RESET     "\033[0m"
+
+/* Line type for coloring */
+enum line_type {
+	LINE_FILE = 0,
+	LINE_HEADER,
+	LINE_HUNK,
+	LINE_ADDED,
+	LINE_REMOVED,
+};
+
+/* ANSI color codes for diff output */
+static char*color_codes[] = {
```

Comment:
Change to `static const char *const color_codes[] = {`

The elements of this array are `const char` pointers (hence `static char *color_codes[]` ==> `static const char *color_codes[]`), and the array itself should be const (hence `static const char *color_codes[]` ==> `static const char *const color_codes[]`).